### PR TITLE
Add substr function description document

### PR DESCRIPTION
**Pull request by dongweizhao**
# Proposed changes

Issue Number: close #xxx

## Problem Summary:

Add substr function description

## Checklist(Required)

1. Does it affect the original behavior: No
2. Has unit tests been added: No
3. Has document been added or modified: Yes
4. Does it need to update dependencies: No
5. Are there any changes that cannot be rolled back:No

## Further comments

Local tested

### CODE REVIEW

**Comment by luzhijing** [comment]
LGTM

File: docs/en/sql-reference/sql-functions/string-functions/substr.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+### Syntax
+
+`VARCHAR substr(VARCHAR content, INT start, INT length)`
+
+Find a substring, return the part of the string described by the first parameter starting from start and having a length of len. The index of the first letter is 1.
+
+## example
+
+```
+mysql> select substr("Hello doris",2,1);
```
Comment (qidaye) [inline docs/en/sql-reference/sql-functions/string-functions/substr.md:38]:
```suggestion
mysql> select substr("Hello doris", 2, 1);
```

File: docs/en/sql-reference/sql-functions/string-functions/substr.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
++-----------------------------+
+| substr('Hello doris', 2, 1) |
++-----------------------------+
+| e                           |
++-----------------------------+
+
+
+mysql> select substr("Hello doris",1,2);
```
Comment (qidaye) [inline docs/en/sql-reference/sql-functions/string-functions/substr.md:46]:
```suggestion
mysql> select substr("Hello doris", 1, 2);
```

File: docs/zh-CN/sql-reference/sql-functions/string-functions/substr.md [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+### Syntax
+
+`VARCHAR substr(VARCHAR content, INT start, INT length)`
+
+求子字符串，返回第一个参数描述的字符串中从start开始长度为len的部分字符串。首字母的下标为1。
+
+## example
+
+```
+mysql> select substr("Hello doris",2,1);
```
Comment (qidaye) [inline docs/zh-CN/sql-reference/sql-functions/string-functions/substr.md:38]:
```suggestion
mysql> select substr("Hello doris", 2, 1);
```

File: docs/zh-CN/sql-reference/sql-functions/string-functions/substr.md [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
++-----------------------------+
+| substr('Hello doris', 2, 1) |
++-----------------------------+
+| e                           |
++-----------------------------+
+
+
+mysql> select substr("Hello doris",1,2);
```
Comment (qidaye) [inline docs/zh-CN/sql-reference/sql-functions/string-functions/substr.md:46]:
```suggestion
mysql> select substr("Hello doris", 1, 2);
```

**Review by qidaye — COMMENTED** [review]
Attention to the format.
Please format your commit info according to the document [commit-format](https://doris.apache.org/zh-CN/community/commit-format-specification.html)

**Comment by qidaye** [comment]
LGTM

**Review by qidaye — COMMENTED** [review]
LGTM

**Comment by dongweizhao** [comment]
> Attention to the format. Please format your commit info according to the document [commit-format](https://doris.apache.org/zh-CN/community/commit-format-specification.html)

OK, pay attention next time

**Review by morningman — DISMISSED** [review]
LGTM

**Comment by BiteTheDDDDt** [comment]
`substr` have overload `substr(str,pos)`, could you please add this to document?

**Comment by morningman** [comment]
hi @dongweizhao , please resolve the conflict

**Comment by dongweizhao** [comment]
> `substr` have overload `substr(str,pos)`, could you please add this to document?

Yes, I will add it after I verify it @BiteTheDDDDt

**Comment by morrySnow** [comment]
@dongweizhao Are you still working on this PR?

**Comment by jackwener** [comment]
This PR has not been updated for a long time.

There could be many reasons.

If it's caused by there are no maintainers to push or ignore this PR, I'm sorry.

If you contact maintainer or want to continue this PR, feel free to @me or @other maintainers.

**Pull request closed without merging**